client-metrics-reporter/src/main/java/io/strimzi/kafka/metrics/prometheus/common/AbstractReporter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.metrics.prometheus.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+
+/**
+ * Common reporter logic to track metrics that match an allowlist pattern.
+ */
+public abstract class AbstractReporter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractReporter.class);
+
+    private final Map<Object, MetricWrapper> allowedMetrics = new ConcurrentHashMap<>();
```

Comment:
I don't find the naming of `otherMetrics` too clear.   It took me a while to grok what was going on:

You keep track of the otherMetrics so that in the case the allow list is changed, you can start to report on the newly matched metrics.  You only need to do this if the implementation is reconfigurable.

Can I ask for a few more comments (class level)?

Reply:
I agree with Keith, it would be better to have some comments.

Reply:
I renamed `otherMetrics` to `disallowedMetrics`. I also added a few comments to this class to clarify its role. Let me know if some parts are still not clear.